DOCS-#3048: document `modin.backends.pyarrow` module (#3051)
modin-project/modin: docs/flow/modin/backends/pyarrow/query_compiler.rst

PyArrow Query Compiler
""""""""""""""""""""""
:py:class:`~modin.backends.pyarrow.query_compiler.PyarrowQueryCompiler` is responsible for compiling efficient
DataFrame algebra queries for the :doc:`PyarrowOnRayFrame </flow/modin/experimental/engines/pyarrow_on_ray>`, 

> the such frames that are backed by ``pyarrow.Table`` objects.

Reviewer: ```suggestion
the frames for which are backed by ``pyarrow.Table`` objects.
```

Author: this probably should be "the frames which are backed by ``pyarrow.Table`` objects." (without *for*).
Changed it

PyArrow query compiler
""""""""""""""""""""""
``PyarrowQueryCompiler`` is responsible for compiling efficient DataFrame algebra queries for the
:doc:`PyarrowOnRayFrame </flow/modin/experimental/engines/pyarrow_on_ray>`, the such frames that
are backed by ``pyarrow.Table`` objects.

> Each ``PyarrowQueryCompiler`` contains an instance of ``PyarrowOnRayFrame`` which it queries to get the result.

Reviewer 2: ```suggestion
Each :py:class:`modin.backends.pyarrow.query_compiler.PyarrowQueryCompiler` contains an instance of :py:class:`modin.experimental.engines.pyarrow_on_ray.frame.data.PyarrowOnRayFrame` which it queries to get the result.
```
